scalapb-runtime/src/main/scala/scalapb/UnknownFieldSet.scala

```diff
@@ -89,7 +89,7 @@ object UnknownFieldSet {
           lengthDelimited = lengthDelimited.result()
         )
 
-      def parseField(tag: Int, input: CodedInputStream) = {
+      def parseField(tag: Int, input: CodedInputStream): this.type = {
```

Comment:
Curious, why `this.type` and not `Builder`?

Reply:
The "unused warnings" don't warn in this case because `this.type` is not a value that is produced.

I made this change first (because it is a usual change; StringBuilder got this.type returns not long ago) but after making the other edit, maybe ignoring the value and adding nowarn is more general and doesn't mess with signatures.

Reply:
And now I see it doesn't work on 2.12 anyway, which does have nowarn support IIRC. (I did run 2_13 e2e; oh wait, I thought warnings weren't on for 2.12?)

Reply:
If we use `this.type` to achieve what `nowarn` does, then I think we should just use `@nowarn` - generally keep things in simplest form and minimizing change. I'm ok with turning off warnings for 2.12 at this stage...

Reply:
OK I'll update with nowarn, thanks.